src/faction_camp.cpp

```diff
@@ -190,458 +192,477 @@ void talk_function::camp_missions( mission_data &mission_key, npc &p )
+                                  "to rapidly dismantle large vehicles, and a forge provides the "
+                                  "resources to make charcoal.  \n \n"
+                                  "NOTE: Actions available through expansions are located in "
+                                  "separate tabs of the Camp Manager window.  \n \n"
+                                  "Risk: None\n"
+                                  "Time: 3 Hours \n"
+                                  "Positions: %d/1\n" ), npc_list.size() );
+        mission_key.add_start( "Expand Base", _( "Expand Base" ), "", entry, npc_list.empty() );
         if( !npc_list.empty() ) {
             entry = _( "Surveying for expansion...\n" );
             bool avail = false;
             for( auto &elem : npc_list ) {
                 avail |= update_time_left( entry, elem );
             }
             entry = entry + _( "\n \nDo you wish to bring your allies back into your party?\n" );
-            mission_key.text["Recover Surveyor"] = entry;
-            mission_key.push( "Recover Surveyor", _( "Recover Surveyor" ), "", true, avail );
+            mission_key.add_return( "Recover Surveyor", _( "Recover Surveyor" ), "", entry, avail );
         }
     }
 
     if( bcp->has_level( camp_ctr, 5, base_dir ) ) {
-        std::vector<std::shared_ptr<npc>> npc_list = companion_list( p, "_faction_camp_cut_log" );
-        mission_key.text["Cut Logs"] = string_format( _( "Notes:\n"
-                                       "Send a companion to a nearby forest to cut logs.\n \n"
-                                       "Skill used: fabrication\n"
-                                       "Difficulty: 1 \n"
-                                       "Effects:\n"
-                                       "> 50%% of trees/trunks at the forest position will be cut down.\n"
-                                       "> 100%% of total material will be brought back.\n"
-                                       "> Repeatable with diminishing returns.\n \n"
-                                       "Risk: None\n"
-                                       "Time: 6 Hour Base + Travel Time + Cutting Time\n"
-                                       "Positions: %d/1\n" ), npc_list.size() );
-        mission_key.push( "Cut Logs", _( "Cut Logs" ), "", false, npc_list.empty() );
+        comp_list npc_list = companion_list( p, "_faction_camp_cut_log" );
+        entry = string_format( _( "Notes:\n"
+                                  "Send a companion to a nearby forest to cut logs.\n \n"
+                                  "Skill used: fabrication\n"
+                                  "Difficulty: 1 \n"
+                                  "Effects:\n"
+                                  "> 50%% of trees/trunks at the forest position will be cut down.\n"
+                                  "> 100%% of total material will be brought back.\n"
+                                  "> Repeatable with diminishing returns.\n \n"
```

Comment:
Might be worth mentioning here that the forest will be cleared to a field eventually? could copy some of the text from clearcut.

Reply:
Fixed.